frontend/src/api/httpClientWithInterceptors.tsx

```diff
@@ -0,0 +1,26 @@
+
+import {HttpClient, IHttpRequestOptions, defaultHttpRequestOptions} from './HttpClient'
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type AbstractApiData = any
+
+export class HttpClientWithInterceptors extends HttpClient {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  //@ts-ignore
```

Comment:
- Do not use `@ts-ignore` unless it some special case.
 - execute should be an async method
 - change `execute` of both `HttpClient` and `HttpClientWithInterceptors` from `private` to `protected`
 - Let it work almost identical as `HttpClient`. Replace with fake response only if exact URL (in our case - URL to change email) is called.